### PR TITLE
add python solution to usaco-788

### PR DESCRIPTION
**Pull request by 876pol**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/silver/usaco-788.mdx [hunk 1]
```diff
@@ -145,4 +145,56 @@ public class MooTube {
...
+
+sys.setrecursionlimit(int(1e9))
+
+# Variables to store graph data and visited nodes
+neighbors = []
+visited = []
+threshold = 0
+num_reachable = 0
+
+# Function to search all vertices that can be reached through the current vertex v
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-788.mdx:165]: use docstring

File: solutions/silver/usaco-788.mdx [hunk 2]
```diff
@@ -145,4 +145,56 @@ public class MooTube {
...
+```py
+import sys
+
+sys.setrecursionlimit(int(1e9))
+
+# Variables to store graph data and visited nodes
+neighbors = []
+visited = []
+threshold = 0
+num_reachable = 0
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-788.mdx:163]: you can put these variables above the `with` blocks imo
also `threshold` here is unnecessary

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]
:fire:

**Review by ryanchou-dev — APPROVED** [review]
fire

**Pull request merged**